components/p3a/nitro_utils/attestation.cc

```diff
@@ -89,30 +104,52 @@ bool VerifyUserDataKey(scoped_refptr<net::X509Certificate> server_cert,
   CHECK(server_cert);
 
   const auto user_data_it = cose_map.find(cbor::Value("user_data"));
-  if (user_data_it == cose_map.end() || !user_data_it->second.is_bytestring() ||
-      user_data_it->second.GetBytestring().size() < kUserDataMinLength) {
-    LOG(ERROR) << "Nitro verification: user data is missing or is not bstr "
-               << "or is not at least " << kUserDataMinLength << " bytes";
-    return false;
-  }
-  if (memcmp(user_data_it->second.GetBytestring().data(), kHashPrefix,
-             kHashPrefixLength) != 0) {
-    LOG(ERROR) << "Nitro verification: user data is missing sha256 hash prefix";
+  if (user_data_it == cose_map.end() || !user_data_it->second.is_bytestring()) {
+    LOG(ERROR) << "Nitro verification: user data is missing or is not bstr";
     return false;
   }
+  const auto user_data_bytes = user_data_it->second.GetBytestring();
+
+  // Fingerprint of the connection TLS cert to compare against.
   const net::SHA256HashValue server_cert_fp =
       net::X509Certificate::CalculateFingerprint256(server_cert->cert_buffer());
-  if (memcmp(server_cert_fp.data,
-             user_data_it->second.GetBytestring().data() + kHashPrefixLength,
-             kSHA256HashLength) != 0) {
-    LOG(ERROR)
-        << "Nitro verification: server cert fp does not match user data fp, "
-        << "user data = "
-        << base::HexEncode(user_data_it->second.GetBytestring())
-        << ", server cert fp = " << base::HexEncode(server_cert_fp.data);
-    return false;
+
+  // The hashes in the  old and new user_data schemes have incommensurate
+  // lengths, so use the total length to distinguish between them.
+  if (user_data_bytes.size() == kUserDataOldLength) {
+    if (memcmp(user_data_bytes.data(), kHashPrefix, kHashPrefixLength) != 0) {
```

Comment:
Unless this is executed in a secure-element/separated context (a local/remote environment), the user can already read the data.
Are any page/tab context able to access an oracle out of this comparison? If the answer is yes it is indeed a local/remote environment.

I think if we have any doubt about if this will have a oracle or not, it is worth changing to `crypto::SecureMemEqual`. The cost should be negligible.